src/rules/valid-title.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  createRule,
+  getNodeName,
+  getStringValue,
+  isDescribe,
+  isStringNode,
+  isTestCase,
+} from './utils';
+
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+const trimFXprefix = (word: string) =>
+  ['f', 'x'].includes(word.charAt(0)) ? word.substr(1) : word;
+
+const getNodeTitle = (node: TSESTree.CallExpression): string | null => {
+  const [argument] = node.arguments;
+
+  return isStringNode(argument) ? getStringValue(argument) : null;
+};
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Enforce valid titles',
+      recommended: false,
+    },
+    messages: {
+      duplicatePrefix: 'should not have duplicate prefix',
+      accidentalSpace: 'should not have space in the beginning',
+    },
+    type: 'suggestion',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (!isDescribe(node) && !isTestCase(node)) return;
+
+        const title = getNodeTitle(node);
+        if (!title) return;
+
+        if (title.trimLeft().length !== title.length) {
```

Comment:
You can actually just compare the two strings - by comparing `length`, *in theory* if the string somehow changes it could still pass:

```
String.prototype.trimLeft = function() { return this.toUpperCase(); }

const myString = ' hello world';

console.log('are strings equal?', myString.trimLeft().length === myString.length ? 'yes' : 'no');
```

This is o/c a really horrible thing that no one should ever be doing, and it saves us a few characters, but just wanted to point it out as it's useful to keep in mind for other conditionals 🙂